src/vs/workbench/browser/parts/editor/tabsTitleControl.ts

```diff
@@ -222,6 +223,55 @@ export class TabsTitleControl extends TitleControl {
 		});
 	}
 
+	private updateTabSizing(): void {
+		if (this.tabsContainer) {
+			const options = this.accessor.partOptions;
+			if (options.tabSizing === 'fixed') {
+				this.tabsContainer.style.setProperty('--tab-sizing-fixed-max-width', `${options.tabSizingFixedMaxWidth}px`);
+				this.tabsContainer.onmouseenter = () => this.setTabsWidthFixed();
+				this.tabsContainer.onmouseleave = () => this.resetTabsWidthFixed();
+			} else {
+				this.tabsContainer.style.removeProperty('--tab-sizing-fixed-max-width');
+				this.tabsContainer.onmouseenter = null;
+				this.tabsContainer.onmouseleave = null;
+				this.resetTabsWidthFixed();
+			}
+		}
+	}
+
+	private setTabsWidthFixed() {
+		// to prevent disturbing effects (especially with wrapping tabs)
+		// when tabs are being fixed quickly after they were relaxed,
+		// snap the tabs into their normal position before we fix their width
+		this.forEachTab((editor, index, tabContainer, tabLabelWidget, tabLabel, tabActionBar) => {
+			tabContainer.classList.add('stop-transition');
+		});
+		void this.tabsContainer?.offsetWidth; // force layout
+
+		this.forEachTab((editor, index, tabContainer, tabLabelWidget, tabLabel, tabActionBar) => {
+			// adjust width so the last tab doesn't wrap onto the next line due to tiny rounding errors
+			// introduced by fixed sizing just overflowing the tab bar's width
+			const { width } = tabContainer.getBoundingClientRect();
+			const adjustedWidth = this.isTabLastInWrappingRow(tabContainer) ? width - 0.1 : width;
```

Comment:
This seems duplicate to me: every last tab in a row when wrapping is on already has a class name `last-in-row` from the layout call here:

https://github.com/microsoft/vscode/blob/6f9263ffade3b77430c7f60228d58a3d19fa0f3c/src/vs/workbench/browser/parts/editor/tabsTitleControl.ts#L1758-L1760

Reply:
thanks, I didn't realize that was already there